build/dsc/templates/dsc_type.rb.erb

```diff
@@ -4,6 +4,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
   require Pathname.new(__FILE__).dirname + '../../' + 'puppet/type/base_dsc'
 
   provide :powershell, :parent => Puppet::Type.type(:base_dsc).provider(:powershell) do
+    confine :powershell_major_version => '5'
```

Comment:
Users are still allowed to use `lcm_config` if they want, sure.

The sentiment here though is that we're trying to set a minimum allowable version (not picking an explicit version). My main concern is ensuring end users are at least on the supported production preview release of WMF5, which is what we test against... and not some prior release of WMF5.

I think it's also nice to be inline with our documentation, which states `lcm_config` is optional.  On earlier WMF5 releases, `lcm_config` is mandatory.